### DISCUSSION TITLE
How is the `lockfileVersion` in the pnpm-lock.yaml determined?

### DISCUSSION THREAD

**Discussion by Aghassi** [pnpm/pnpm#6857] [Q&A] [answered]
I'm asking this question because of an issue I hit here https://github.com/aspect-build/rules_js/issues/1174

TL;DR is that 8.6.0 generated `lockfileVersion` of `6.1` and 8.6.10 generated `lockfileVersion` of `6.0`

**Comment by zkochan** [comment, marked as answer]
Sometimes there are some new fields in the lockfile but otherwise the lockfile is backward compatible, so it can be used with previous versions of pnpm. In this case, a new `settings` field was added to the lockfile. However, we had to change the lockfile version back as there is a bug in pnpm v8. It thinks that `6.1` is a breaking change.

**Reply by KSXGitHub** [reply]
Normally, spec version do not contain minor ver, especially ones that are generated by machine. Examples: `yarn.lock`, `package-lock.json`.

**Reply by Aghassi** [reply]
Thanks for the explanation! If I had not had a lockfile up to date check on CI I wouldn't have hit this. It is unfortunate a patch version caused this regression, but I'm glad it was fixed. Also helped us remember to enforce `corepack` and encode your pnpm versions

**Reply by EndangeredMassa** [reply]
Lockfiles are not necessarily backwards compatible, depending on how exactly you interpret term. Note that some pnpm versions cannot parse old lockfile versions.
See: https://github.com/pnpm/spec/issues/7

**Reply by zkochan** [reply]
in the future lockfile version will equal the pnpm version in which it got introduced. The current lockfileVersion is 9.0 as it was released with pnpm v9.0

**Reply by helloint** [reply]
@zkochan will it related to the node version?
I asked because I want to know if have nvm installed to manage different node in different project. Then to avoid install pnpm on each node env, I want to install pnpm on Homebrew. I noticed pnpm needs node as well, so I also installed node on Homebrew, the latest version is node 22. This way, on every node/npm projects, I will have pnpm available from Homebrew. I hope this is fine to go.
